recipe/meta.yaml

```
@@ -24,6 +24,8 @@ build:
         --buildtype=release -Denable_tests=false
     - meson compile -C _build
     - meson install -C _build --no-rebuild
+  run_exports:
+    - {{ pin_subpackage( name|lower, max_pin='x.x.x') }}
```

```suggestion
    - {{ pin_subpackage( name|lower, max_pin='x.x') }}
```

ABI Laboratory shows that they seem to [stick to `x.x`](https://abi-laboratory.pro/?view=timeline&l=dav1d). Do you think we can relax it a bit?

---

I am going to go ahead and merge, hoping I can get the `avif` rebuild in soon too :)

---

Since this is a major release, with 1.0.0, I'd expect more stable ABI, but for now there are no more versions in conda-forge, so this is safe.

---

The pinning to the patch level is the safest option which is why I chose it. If you want to relax the pinning, I suggest contacting the upstream maintainers to get an explicit promise of ABI/API compatibility.

Related: What do you think about putting the ABI major version in the build string? i.e.
`david=1.0.0=b6hsdfbiwd` where the SO major version is 6. Then we could pin to the API and ABI separately.

```yaml
run_export:
  - {{ pin_subpackage( name|lower ) }} b{{ so_name_major }}h*
```

This would be templated out to 

```yaml
run_exports:
  - david >=1.0.0,<2 b6h*
```

or maybe

```yaml
run_exports:
  - david >=1.0.0,<2
  - david * b6h*
```

So then the run export handles both API and ABI. The ABI version is between "b" and "h" because if it was just `b6*`, then we could run into issues when we reach `b60`.

The more C++ projects that I become the maintainer of, the more I run into the problem that conda doesn't have a built in way for tracking ABI separately from API (because conda's roots are in python where you don't worry about ABI).

---

I like this idea! I was also thinking about how API contracts don't always follow ABI contracts and we are forced to find a way to track it weirdly. That said, I feel this conversation could benefit from having a bigger audience, and eventually become a CFEP itself. What about an issue in `conda-forge/conda-forge.github.io` first?

---

Is that the better way to introduce a CFEP? I was just going to create a CFEP directly, but my other one hasn't created much discusssion.